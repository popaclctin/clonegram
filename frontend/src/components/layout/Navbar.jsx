import React from 'react';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faHouse,
  faSquarePlus,
  faCircleUser,
} from '@fortawesome/free-solid-svg-icons';

function Navbar() {
  return (
    <nav className='header__nav'>
      <ul>
        <li>
          <NavLink to='/'>
            <FontAwesomeIcon icon={faHouse} size='xl' />
          </NavLink>
        </li>
        <li>
          <NavLink to='/post/create'>
            <FontAwesomeIcon icon={faSquarePlus} size='xl' />
          </NavLink>
        </li>
        <li>
          <FontAwesomeIcon icon={faCircleUser} size='xl' />
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
