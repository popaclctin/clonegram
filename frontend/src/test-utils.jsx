import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setupStore } from './store';
import { setupListeners } from '@reduxjs/toolkit/dist/query';

import { BrowserRouter } from 'react-router-dom';

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) {
  setupListeners(store.dispatch);

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
