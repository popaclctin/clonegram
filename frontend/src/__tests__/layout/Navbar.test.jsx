import React from 'react';
import Navbar from '../../components/layout/Navbar';
import { renderWithProviders } from '../../test-utils';

describe('<Navbar />', () => {
  it('renders correctly', () => {
    const { asFragment } = renderWithProviders(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
