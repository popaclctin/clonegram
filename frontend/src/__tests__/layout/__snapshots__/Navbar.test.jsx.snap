// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="header__nav"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-house fa-xl "
            data-icon="house"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
      <li>
        <button
          class="header__nav__createPostBtn"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-square-plus fa-xl "
            data-icon="square-plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zM200 344V280H136c-13.3 0-24-10.7-24-24s10.7-24 24-24h64V168c0-13.3 10.7-24 24-24s24 10.7 24 24v64h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H248v64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
              fill="currentColor"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          class="header__nav__userBtn"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-user fa-xl "
            data-icon="circle-user"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M399 384.2C376.9 345.8 335.4 320 288 320H224c-47.4 0-88.9 25.8-111 64.2c35.2 39.2 86.2 63.8 143 63.8s107.8-24.7 143-63.8zM512 256c0 141.4-114.6 256-256 256S0 397.4 0 256S114.6 0 256 0S512 114.6 512 256zM256 272c39.8 0 72-32.2 72-72s-32.2-72-72-72s-72 32.2-72 72s32.2 72 72 72z"
              fill="currentColor"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
